util/argo/audit_logger.go

```diff
@@ -68,7 +68,7 @@ func (l *AuditLogger) logEvent(objMeta ObjectRef, gvk schema.GroupVersionKind, i
 		ObjectMeta: metav1.ObjectMeta{
 			Name:        fmt.Sprintf("%v.%x", objMeta.Name, t.UnixNano()),
 			Labels:      eventLabels,
-			Annotations: logFields,
+			Annotations: eventAnnotations,
```

Comment:
If we want to start adding information into logFields which won't make their way into the event, we need to split this parameter into two.

This is why the PR has exploded in size.